File: Level-1/67.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        
        HashMap<Character, Integer> map = new HashMap<>();
        
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if(map.containsKey(c)) {
                int idx = map.get(c);
                answer[i] = i - idx;
            }else {
                answer[i] = -1;
            }
            map.put(c, i);
        }
        
        return answer;
    }
}

File: Level-1/67.js
function solution(s) {
    let answer = [];
    const map = new Map();
    
    for(let i = 0; i < s.length; i++) {
        const c = s[i];
        if(map.has(c)) {
            answer.push(i - map.get(c));
        }else {
            answer.push(-1);
        }
        map.set(c, i);
    }
    
    return answer;
}
